dali/pipeline/data/tensor_list_gpu.cu

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/pipeline/data/tensor_list.h"
+
+namespace dali {
+
+__global__ void CopyKernel(uint8_t *dst, const uint8_t *src, int64_t n) {
+  for (int64_t i = blockIdx.x * blockDim.x + threadIdx.x; i < n; i += blockDim.x * gridDim.x) {
+    dst[i] = src[i];
+  }
+}
+
+void LaunchCopyKernel(void *dst, const void *src, int64_t n, cudaStream_t stream) {
+  size_t block = cuda_min(n, 1024l);
+  size_t grid = std::min(32l, div_ceil(n, block));
+  CopyKernel<<<grid, block, 0, stream>>>(reinterpret_cast<uint8_t*>(dst), reinterpret_cast<const uint8_t*>(src), n);
+  cudaGetLastError();
+}
```

Comment:
How about making this templated?
```c++
template<typename T>
void LaunchCopyKernel(T *dst, const T *src, int64_t n, cudaStream_t stream) {
  size_t block = cuda_min(n*sizeof(T), 1024l);
  size_t grid = std::min(32l, div_ceil(n*sizeof(T), block));
  CopyKernel<<<grid, block, 0, stream>>>(reinterpret_cast<uint8_t*>(dst), reinterpret_cast<const uint8_t*>(src), n);
  cudaGetLastError();
}
```

Reply:
I don't see the use. What I can do is rename `n` to `nbytes` for clarification

Reply:
`nbytes` would be better ;)